src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -120,11 +123,11 @@ public void Write(CompilerData compilerData, ObjectModuleDetails omDetails)
             string library = omDetails?.Library;
             if (this.appInsightsRegistered)
             {
-                s_telemetryClient.TrackEvent(CompilerEventName, properties: new Dictionary<string, string>
+                string commandLineId = Guid.NewGuid().ToString();
```

Comment:
>commandLineId [](http://example.com/codeflow?start=23&length=13)

create only when you are sure that you have a commandLine.
otherwise, you will create something that will be discarted